src/svgp.jl

```diff
@@ -79,5 +79,47 @@ end
 
 inducing_points(f::ApproxPosteriorGP{<:SVGP}) = f.approx.fz.x
 
-_chol_cov(q::AbstractMvNormal) = cholesky(Symmetric(cov(q)))
-_chol_cov(q::MvNormal) = cholesky(q.Σ)
+## Pathwise posterior sampling
+
+struct PosteriorFunctionSamples
+    num_samples
+    prior  # prior(x) returns an approx prior sample
+    update  # update(x) computes the required pathwise update
+end
+
+(f::PosteriorFunctionSamples)(x) = f([x])
+(f::PosteriorFunctionSamples)(x::AbstractVector) = f.prior(x) + f.update(x)
```

Comment:
Do we have a notion of `output_dims`? i.e. since we take the extended-input-domain view of MOGPs throughout the ecosystem, shouldn't it always be possible to think in terms of a matrix of size `(num_points, num_samples)`?

> Alternatively, pathwise_sample could return a vector of functions (one per sample)?

My inclination would be not to define this function like this, but to instead do something like
```julia
map(f::PosteriorFunctionSamples, x::AbstractVector) = map(f.prior, x) + map(f.update, x)
```
You could then represent the output with a `ColVecs` or something.